### PR TITLE
Add new token group "space" to replace spacing in v11

### PR DESCRIPTION
**Pull request by alex-page**
- [x] Update polaris.shopify.com docs
- [x] Add new token group `space`
- [x] Update types to remove `Spacing` and duplicate `SpacingSpace`
- [x] Replace usage of `spacing` with `space` in tsx
- [x] Does not touch scss as the token names are the exact same

### CODE REVIEW

File: polaris-for-vscode/src/server.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ const groupedCompletionItemPatterns: Omit<
   font: /font|line-height/,
   motion: /animation/,
   shadow: /shadow/,
-  spacing: /margin|padding|gap|top|left|right|bottom/,
+  space: /margin|padding|gap|top|left|right|bottom/,
```
Comment (lgriffee) [inline polaris-for-vscode/src/server.ts:91]: I'd prefer to handle these changes in https://github.com/Shopify/polaris/pull/9010

File: polaris.shopify.com/content/components/layout-and-structure/bleed.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ examples:
   - fileName: bleed-specific-direction.tsx
     title: Specific direction
     description: >-
-      Negative margins can be added in a specific direction using the [Spacing tokens](https://polaris.shopify.com/tokens/spacing).
+      Negative margins can be added in a specific direction using the [Space tokens](https://polaris.shopify.com/tokens/space).
```
Comment (lgriffee) [inline polaris.shopify.com/content/components/layout-and-structure/bleed.md:22]: I'd prefer to handle this change in https://github.com/Shopify/polaris/pull/9006

File: polaris.shopify.com/content/tools/stylelint-polaris/index.md [hunk 3]
```diff
@@ -90,24 +90,24 @@ There are over 40 rules configured in Stylelint Polaris to help you avoid errors
 - [Media queries](/tools/stylelint-polaris/rules#media-queries)
 - [Motion](/tools/stylelint-polaris/rules#motion)
 - [Shape](/tools/stylelint-polaris/rules#shape)
-- [Spacing](/tools/stylelint-polaris/rules#spacing)
+- [Space](/tools/stylelint-polaris/rules#space)
```
Comment (lgriffee) [inline polaris.shopify.com/content/tools/stylelint-polaris/index.md:93]: I'd prefer to handle this change in https://github.com/Shopify/polaris/pull/9006

File: polaris.shopify.com/pages/tokens/spacing.tsx [hunk 4]
```diff
@@ -6,8 +6,8 @@ import PageMeta from '../../src/components/PageMeta';
 const Components: NextPage = () => {
   return (
     <>
-      <PageMeta title="Spacing tokens" />
-      <TokensPage tokenGroup={'spacing'} />
+      <PageMeta title="Space tokens" />
+      <TokensPage tokenGroup={'space'} />
```
Comment (lgriffee) [inline polaris.shopify.com/pages/tokens/spacing.tsx:10]: I'd prefer to handle this change in https://github.com/Shopify/polaris/pull/9006

File: polaris.shopify.com/src/components/TokensPage/TokensPage.tsx [hunk 5]
```diff
@@ -16,7 +16,7 @@ interface Props {
     | 'motion'
     | 'shadow'
     | 'shape'
-    | 'spacing'
+    | 'space'
```
Comment (lgriffee) [inline polaris.shopify.com/src/components/TokensPage/TokensPage.tsx:19]: I'd prefer to handle this change in https://github.com/Shopify/polaris/pull/9006

File: .changeset/eighty-maps-sneeze.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-tokens': minor
```
Comment (lgriffee) [inline .changeset/eighty-maps-sneeze.md:2]: Do we also need an entry here for polaris-react?

Comment (alex-page) [reply .changeset/eighty-maps-sneeze.md:2]: Yes. That is a good call!

File: polaris-for-vscode/src/server.ts (hunk 1 above)
Comment (lgriffee) [reply polaris-for-vscode/src/server.ts:91]: Actually going to just add those updates to this branch instead!

**Review by lgriffee — APPROVED** [review]
Okay I added some additional spacing updates and I think you should be good to go! Also noticed two `polaris.shopify.com/scripts/tmp` files sneak in and wasn't sure if that was intentional?

**Comment by alex-page** [comment]
`polaris.shopify.com/scripts/tmp` SNEAKY! I'll be sure to eradicate them. Thanks @lgriffee

File: polaris.shopify.com/content/tools/stylelint-polaris/rules/space-global-disallowed-list.md [hunk 7]
```diff
@@ -1,10 +1,10 @@
 ---
-title: spacing/global-disallowed-list
+title: space/global-disallowed-list
 description: Disallows use of legacy spacing Sass APIs.
 keywords:
   - stylelint
-  - spacing
-  - spacing rules
+  - space
+  - space
```
Comment (alex-page) [inline polaris.shopify.com/content/tools/stylelint-polaris/rules/space-global-disallowed-list.md]:
```suggestion
  - space rules
```

**Comment by lgriffee** [comment]
Going to add a redirect for the new polaris.shopify.com page in `next.config.js`. After this ships just want to make a note we need to remember to remove the spacing token group from the [PR in v11](https://github.com/Shopify/polaris/pull/8374).

**Comment by lgriffee** [comment]
Oh we also need redirects for all the stylelint pages! Going to add that too before this ships 👍🏻

**Comment by lgriffee** [comment]
Redirects aren't working in localhost. Looks like I forgot to add the new groups. Don't ship this yet.

**Review by aaronccasanova — APPROVED** [review]

**Comment by lgriffee** [comment]
Fixed the redirects so they work on localhost! Also it looks like anchor link redirects don't work so I removed it.

**Pull request merged**